src/test/java/seedu/address/logic/commands/EventCommandTest.java

```diff
@@ -0,0 +1,294 @@
+        ModelStubAcceptingEventAdded modelStub = new ModelStubAcceptingEventAdded();
+        Event validEvent = new EventBuilder().buildWithNoAttendees();
+        CommandResult commandResult =
+                new EventCommand(DEFAULT_NAME, DEFAULT_DATE, DEFAULT_INDEXES).execute(modelStub);
+
+        assertEquals(String.format(EventCommand.MESSAGE_SUCCESS, Messages.formatEvent(validEvent)),
+                commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validEvent), modelStub.eventsAdded);
+    }
+
+    @Test
+    public void execute_duplicateEvent_throwsCommandException() {
+        Event validEvent = new EventBuilder().buildWithNoAttendees();
+        EventCommand eventCommand = new EventCommand(DEFAULT_NAME, DEFAULT_DATE, DEFAULT_INDEXES);
+        ModelStub modelStub = new ModelStubWithEvent(validEvent);
+
+        assertThrows(CommandException.class,
+                EventCommand.MESSAGE_DUPLICATE_EVENT, () -> eventCommand.execute(modelStub));
+    }
+
+    @Test
+    public void execute_outOfBoundsIndex_throwsCommandException() {
+        Index index = Index.fromOneBased(SampleDataUtil.ADDRESSBOOK_SIZE + 1);
+        Set<Index> indexes = new HashSet<>();
+        indexes.add(index);
+        EventCommand eventCommand = new EventCommand(DEFAULT_NAME, DEFAULT_DATE, indexes);
+
+        ModelStubAcceptingEventAdded modelStub = new ModelStubAcceptingEventAdded();
+        assertThrows(CommandException.class,
+                Messages.MESSAGE_ATTENDEE_NOT_FOUND, () -> eventCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        EventCommand eventCommand1 = new EventCommand("event1", DEFAULT_DATE, DEFAULT_INDEXES);
+        EventCommand eventCommand2 = new EventCommand("event2", DEFAULT_DATE, DEFAULT_INDEXES);
+
+        // same object -> returns true
+        assertTrue(eventCommand1.equals(eventCommand1));
+
+        // same values -> returns true
+        EventCommand eventCommandCopy = new EventCommand("event1", DEFAULT_DATE, DEFAULT_INDEXES);
+        assertTrue(eventCommand1.equals(eventCommandCopy));
+
+        // different types -> returns false
+        assertFalse(eventCommand1.equals(1));
+
+        // null -> returns false
+        assertFalse(eventCommand1.equals(null));
+
+        // different person -> returns false
+        assertFalse(eventCommand1.equals(eventCommand2));
+    }
+
+    @Test
+    public void toStringMethod() {
+        EventCommand eventCommand = new EventCommand(DEFAULT_NAME, DEFAULT_DATE, DEFAULT_INDEXES);
+        String expected = EventCommand.class.getCanonicalName()
+                + "{eventName=" + DEFAULT_NAME + ", "
+                + "eventDate=" + DEFAULT_DATE + ", "
+                + "attendeeIndexes=" + DEFAULT_INDEXES + "}";
+        assertEquals(expected, eventCommand.toString());
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
```

Comment:
Maybe useful to move this into a separate file and make it a public utility class so other test classes can extend the stub

Reply:
Agreed. Changes have been implemented